bgpd/bgp_flowspec_vty.c

```diff
@@ -0,0 +1,384 @@
+/* BGP FlowSpec VTY
+ * Copyright (C) 2018 6WIND
+ *
+ * This file is part of FRR.
+ *
+ * FRRouting is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2, or (at your option) any
+ * later version.
+ *
+ * FRRouting is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; see the file COPYING; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include <zebra.h>
+#include "command.h"
+
+#include "bgpd/bgpd.h"
+#include "bgpd/bgp_table.h"
+#include "bgpd/bgp_attr.h"
+#include "bgpd/bgp_ecommunity.h"
+#include "bgpd/bgp_vty.h"
+#include "bgpd/bgp_route.h"
+#include "bgpd/bgp_aspath.h"
+#include "bgpd/bgp_flowspec.h"
+#include "bgpd/bgp_flowspec_util.h"
+#include "bgpd/bgp_flowspec_private.h"
+#include "bgpd/bgp_debug.h"
+
+/* Local Structures and variables declarations
+ * This code block hosts the struct declared that host the flowspec rules
+ * as well as some structure used to convert to stringx
+ */
+
+static const struct message bgp_flowspec_display_large[] = {
+	{FLOWSPEC_DEST_PREFIX, "Destination Address"},
+	{FLOWSPEC_SRC_PREFIX, "Source Address"},
+	{FLOWSPEC_IP_PROTOCOL, "IP Protocol"},
+	{FLOWSPEC_PORT, "Port"},
+	{FLOWSPEC_DEST_PORT, "Destination Port"},
+	{FLOWSPEC_SRC_PORT, "Source Port"},
+	{FLOWSPEC_ICMP_TYPE, "ICMP Type"},
+	{FLOWSPEC_ICMP_CODE, "ICMP Code"},
+	{FLOWSPEC_TCP_FLAGS, "TCP Flags"},
+	{FLOWSPEC_PKT_LEN, "Packet Length"},
+	{FLOWSPEC_DSCP, "DSCP field"},
+	{FLOWSPEC_FRAGMENT, "Packet Fragment"},
+	{0}
+};
+
+static const struct message bgp_flowspec_display_min[] = {
+	{FLOWSPEC_DEST_PREFIX, "to"},
+	{FLOWSPEC_SRC_PREFIX, "from"},
+	{FLOWSPEC_IP_PROTOCOL, "proto"},
+	{FLOWSPEC_PORT, "port"},
+	{FLOWSPEC_DEST_PORT, "dstp"},
+	{FLOWSPEC_SRC_PORT, "srcp"},
+	{FLOWSPEC_ICMP_TYPE, "type"},
+	{FLOWSPEC_ICMP_CODE, "code"},
+	{FLOWSPEC_TCP_FLAGS, "flags"},
+	{FLOWSPEC_PKT_LEN, "pktlen"},
+	{FLOWSPEC_DSCP, "dscp"},
+	{FLOWSPEC_FRAGMENT, "pktfrag"},
+	{0}
+};
+
+#define	FS_STRING_UPDATE(count, ptr, format) do {			\
+		if (((format) == NLRI_STRING_FORMAT_DEBUG) && (count)) {\
+			(ptr) += sprintf((ptr), ", "); \
+		} else if (((format) == NLRI_STRING_FORMAT_MIN) && (count)) {\
+			(ptr) += sprintf((ptr), " "); \
+		} \
+		count++;			\
+	} while (0)
```

Comment:
Please fix the formatting on this...

Reply:
What's the status here?  I agree w/ Quentin